deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -197,23 +205,29 @@ private void ingestChildren(ContentContainerObject destObj, Resource parentResc)
 		}
 
 		try {
+			ContentObject obj = null;
 			while (iterator.hasNext()) {
 				Resource childResc = (Resource) iterator.next();
 
 				// Ingest the child according to its object type
 				if (childResc.hasProperty(RDF.type, Cdr.Folder)) {
-					ingestFolder(destObj, parentResc, childResc);
+					obj = ingestFolder(destObj, parentResc, childResc);
 				} else if (childResc.hasProperty(RDF.type, Cdr.Work)) {
-					ingestWork(destObj, parentResc, childResc);
+					obj = ingestWork(destObj, parentResc, childResc);
 				} else if (childResc.hasProperty(RDF.type, Cdr.FileObject)) {
 					if (destObj instanceof WorkObject) {
 						// File object is being added to a work, go ahead
-						ingestFileObject(destObj, parentResc, childResc);
+						obj = ingestFileObject(destObj, parentResc, childResc);
 					} else {
 						// File object is a standalone, so construct a Work around it
-						ingestFileObjectAsWork(destObj, parentResc, childResc);
+						obj= ingestFileObjectAsWork(destObj, parentResc, childResc);
 					}
 				}
+				if (obj != null) {
+					// add premis event for ingestion to each child object
+					addIngestionEventForChild(obj);
```

Comment:
I appreciate the centralized code of this approach, but i think it is going to result in all the premis events being added twice for most objects since addPremisEvents is called twice and it always reads from the same file.  It may be hard to go with this approach without a larger code change